### PR TITLE
fix(DataMapper): Fix schema loading issues

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: https://github.com/KaotoIO/kaoto/issues/2977

When a schema attribute defines its type inline via a child <xs:simpleType> element rather than a type="..." attribute reference, the schema parser could not resolve a type name and crashed. The fix adds null-safety for this case, gracefully falling back to a generic AnyType rather than throwing.

Fixes: https://github.com/KaotoIO/kaoto/issues/2978

When the same global element (with an anonymous complex type) was referenced from many different parent contexts, the schema parser expanded its full subtree independently for each reference, causing exponential object growth. The fix stores a global element's anonymous type as a named fragment just like named types.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved null-safety when processing attributes with inline simple type definitions.
  * Safer handling of anonymous global element references in complex XML schemas to prevent runtime errors.

* **Tests**
  * Added coverage for inline-attribute edge cases.
  * Added performance and correctness tests for a large, deeply nested XML schema with time-budget validation.

* **Chores**
  * Added XML schema test fixtures (small inline-type and large anonymous-ref) for reproducibility.

### CODE REVIEW

File: packages/ui/src/services/xml-schema-document.service.ts [hunk 1]
```diff
@@ -413,7 +413,40 @@ export class XmlSchemaDocumentService {
     fields.push(field);
 
     ownerDoc.totalFieldCount++;
-    XmlSchemaDocumentService.populateSchemaType(ownerDoc, field, resolvedElement.getSchemaType());
+
+    const schemaType = resolvedElement.getSchemaType();
+    if (refTarget != null && schemaType instanceof XmlSchemaComplexType && !schemaType.getQName()) {
```
Comment (lordrip) [inline packages/ui/src/services/xml-schema-document.service.ts]:
```suggestion
    if (refTarget !== null && schemaType instanceof XmlSchemaComplexType && !schemaType.getQName()) {
```

Comment (lordrip) [reply packages/ui/src/services/xml-schema-document.service.ts]: Do we need to check against `undefined` as well?

Comment (igarashitm) [reply packages/ui/src/services/xml-schema-document.service.ts]: Thank you for catching, I thought so first, but double checked and in this case, it would never be `undefined`. pushed the change.
